lib/chef/provider/remote_file/sftp.rb

```diff
@@ -0,0 +1,118 @@
+#
+# Author:: Jesse Campbell (<hikeit@gmail.com>)
+# Copyright:: Copyright 2013-2016, Jesse Campbell
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require "uri"
+require "tempfile"
+require "net/sftp"
+require "chef/provider/remote_file"
+require "chef/file_content_management/tempfile"
+
+class Chef
+  class Provider
+    class RemoteFile
+      class SFTP
+        attr_reader :uri
+        attr_reader :new_resource
+        attr_reader :current_resource
+
+        def initialize(uri, new_resource, current_resource)
+          @uri = uri
+          @new_resource = new_resource
+          @current_resource = current_resource
+          validate_path!
+          validate_userinfo!
+        end
+
+        def hostname
+          @uri.host
+        end
+
+        def port
+          @uri.port
+        end
+
+        def user
+          URI.unescape(uri.user)
+        end
+
+        def filename
+          parse_path if @filename.nil?
+          @filename
+        end
```

Comment:
this looks copied from the ftp class

the thing is that in `#get` you've removed the one reference to this method.  the `filename` in `parse_path` is actually a local variable that gets constructed (hella confusingly -- the ftp class could use a bit of cleanup to make it clearer) and therefore both this `#filename` code and the `#parse_path` code are unused in this class that you've written and could simply be deleted.